dev/io.openliberty.microprofile.telemetry.2.0.logging.internal_fat/fat/src/io/openliberty/microprofile/telemetry/logging/internal_fat/TelemetrySourcesTest.java

```diff
@@ -0,0 +1,289 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.microprofile.telemetry.logging.internal_fat;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.RemoteFile;
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.annotation.ExpectedFFDC;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+@RunWith(FATRunner.class)
+public class TelemetrySourcesTest extends FATServletClient {
+
+    private static Class<?> c = TelemetrySourcesTest.class;
+
+    public static final String SERVER_NAME = "TelemetrySources";
+    public static final String APP_NAME = "MpTelemetryLogApp";
+
+    @Server(SERVER_NAME)
+    public static LibertyServer server;
+
+    public static final String SERVER_XML_ALL_SOURCES = "allSources.xml";
+    public static final String SERVER_XML_EMPTY_SOURCE = "emptySource.xml";
+    public static final String SERVER_XML_NO_SOURCE = "noSource.xml";
+    public static final String SERVER_XML_NO_TELEMETRY_ATTRIBUTE = "noTelemetryAttribute.xml";
+    public static final String SERVER_XML_INVALID_SOURCE = "invalidSource.xml";
+
+    private static final String[] EXPECTED_FAILURES = { "CWMOT5005W", "SRVE0315E", "SRVE0777E" };
+
+    @Before
+    public void testSetup() throws Exception {
+        ShrinkHelper.defaultApp(server, APP_NAME, "io.openliberty.microprofile.telemetry.logging.internal.fat.MpTelemetryLogApp");
+        server.startServer();
+
+    }
+
+    @After
+    public void testTearDown() throws Exception {
+        if (server != null && server.isStarted()) {
+            server.stopServer(EXPECTED_FAILURES);
+        }
+    }
+
+    /*
+     * Test a server with all MPTelemetry sources enabled and ensure message, trace and ffdc logs are bridged.
```

Comment:
Would be good to add how the source attribute looks like in each test case `<mpTelemetry source="message, trace, ffdc"/>`